### PR TITLE
Replaced 'download folder' with 'synchronize', triggering the synchronization in two-ways of all the files in a folder

### PR DESCRIPTION
**Pull request by davivel**

### CODE REVIEW

**Comment by davivel** [comment]
Implemented, but pending of clean-up. I will comment again when development is finished.

**Comment by davivel** [comment]
Clean-up done. Ready to test & review.

cc @purigarcia & @masensio

**Comment by davivel** [comment]
Implements #3 (in manual mode)

**Comment by davivel** [comment]
@purigarcia , @masensio : rebased after last merges into master; ready to go again

File: src/com/owncloud/android/datamodel/FileDataStorageManager.java [hunk 1]
```diff
@@ -1460,44 +1390,6 @@ public void saveSharesInFolder(ArrayList<OCShare> shares, OCFile folder) {
...
-                    getContentProviderClient().delete(  ProviderTableMeta.CONTENT_URI_SHARE, 
-                                                        where,
-                                                        whereArgs);
-
-                } catch (RemoteException e) {
-                    Log_OC.e(TAG, "Exception deleting shares in a folder " + e.getMessage());
-                }
-            }
-            */
-        //}
```
Comment (masensio) [inline src/com/owncloud/android/datamodel/FileDataStorageManager.java]: FileDataStorageManager.java is cleaner. :+1

**Comment by masensio** [comment]
Awesome job!!

Code review done. 
Merge pending until the validation will be finished.

**Comment by davivel** [comment]
Delayed to next release due to problem not detecting some conflicts

**Comment by strugee** [comment]
> Delayed to next release

Is the date for this written down somewhere? I know there's the GitHub Milestones, but they don't say (AFAIK) _when_ releases will be shipped.

I was really, _really_ looking forward to having this feature, and it'd be nice to know when it's expected.

**Comment by davivel** [comment]
MIlestones have a date, though it's not as easy to find as should:

https://github.com/owncloud/android/milestones

In this view dates are listed below the milestone names.

**Comment by strugee** [comment]
@davivel thanks, I appreciate it.

**Comment by rperezb** [comment]
There is a bug here, in case there is a conflict, it´s not possible to fix the conflict 
 STEPS TO REPRODUCE IT
1. Go to the Desktop Client and create a new account (user1)
2. Create a new folder (e.g Hello) on the Desktop cliente
3. Go to the folder (Hello) and create a new File (e.g Hola)
4. Open the file and write something, Save
5. Go to the mobil
6. Go to the mobil with the same account (user1)
7. Waiting to sync
8. Go to the file
9. Download the file
10. Go to the Desktop client
11. Open the file, write and save
12. Go to the mobil (do not upload)
13. Open the file, write and save
14. Select to sync file (You can see File Conflict)
15. If you select the Local Version Option

**Comment by tobiasKaminsky** [comment]
@rperezb There is something missing in your test description, or? Point 15 ends in the middle of the sentence.

**Comment by davivel** [comment]
Bug fixed in #1219.

Aaaaand, please @Dianafg76

**Comment by davivel** [comment]
That's it.

**Comment by Dianafg76** [comment]
I tested with Nexus 9 v6.0 and Android Galaxy S3 v 4 and is working OK

**Comment by masensio** [comment]
Waiting for Travis :)

**Pull request merged**